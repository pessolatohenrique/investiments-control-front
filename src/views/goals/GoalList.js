import React from "react";
import moment from "moment";
import {
  Grid,
  Card,
  CardContent,
  Typography,
  List,
  ListItem,
  ListItemText,
} from "@mui/material";

export function GoalList({ goals }) {
  return (
    <Grid container spacing={2}>
      {[...goals].map((row) => {
        return (
          <Grid item xs={4} key={row.id}>
            <Card data-testid="card-item">
              <CardContent>
                <Typography variant="h6" component="div" color="text.secondary">
                  {row?.name}
                </Typography>
                <List variant="body2">
                  <ListItem>
                    <ListItemText disableTypography>
                      <Typography variant="body2">
                        <strong>Porcentagem atual: </strong>
                        {row?.actual_value_percentage} %
                      </Typography>
                    </ListItemText>
                  </ListItem>
                  <ListItem>
                    <ListItemText disableTypography>
                      <Typography variant="body2">
                        <strong>Porcentagem esperada: </strong>
                        {row?.expected_value_percenage} %
                      </Typography>
                    </ListItemText>
                  </ListItem>
                  <ListItem>
                    <ListItemText disableTypography>
                      <Typography variant="body2">
                        <strong>Valor total: </strong>R$ {row?.total_value}
                      </Typography>
                    </ListItemText>
                  </ListItem>
                  <ListItem>
                    <ListItemText disableTypography>
                      <Typography variant="body2">
                        <strong>Valor atual: </strong>R$ {row?.actual_value}
                      </Typography>
                    </ListItemText>
                  </ListItem>
                  <ListItem>
                    <ListItemText disableTypography>
                      <Typography variant="body2">
                        <strong>Meses restantes: </strong>
                        {row?.remaining_months}
                      </Typography>
                    </ListItemText>
                  </ListItem>
                  <ListItem>
                    <ListItemText disableTypography>
                      <Typography variant="body2">
                        <strong>Meses atuais: </strong>
                        {row?.actual_months}
                      </Typography>
                    </ListItemText>
                  </ListItem>
                  <ListItem>
                    <ListItemText disableTypography>
                      <Typography variant="body2">
                        <strong>Total de meses: </strong>
                        {row?.total_installments}
                      </Typography>
                    </ListItemText>
                  </ListItem>
                  <ListItem>
                    <ListItemText disableTypography>
                      <Typography variant="body2">
                        <strong>Data de in√≠cio: </strong>{" "}
                        {moment(row?.initial_date).format("DD/MM/YYYY")}
                      </Typography>
                    </ListItemText>
                  </ListItem>
                  <ListItem>
                    <ListItemText disableTypography>
                      <Typography variant="body2">
                        <strong>Data de fim: </strong>{" "}
                        {moment(row?.final_date).format("DD/MM/YYYY")}
                      </Typography>
                    </ListItemText>
                  </ListItem>
                </List>
              </CardContent>
            </Card>
          </Grid>
        );
      })}
    </Grid>
  );
}
