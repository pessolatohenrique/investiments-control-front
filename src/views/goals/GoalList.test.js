import moment from "moment";
import { render, screen } from "@testing-library/react";
import { GoalList } from "./GoalList";

const goals = [
  {
    _id: "63507dcba3f4aa14eb41e0ec",
    userId: "630ff7a767ae1e8f6a17631b",
    name: "Imóvel novo",
    total_value: 400000,
    initial_date: "2021-10-01T00:00:00.000Z",
    final_date: "2027-01-01T00:00:00.000Z",
    total_installments: 70,
    monthly_profitability: 0.72,
    invest_in_month: 4000,
    status: true,
    __v: 0,
    actual_value: 22753.4,
    actual_months: 15,
    remaining_months: 47,
    actual_value_percentage: "5.69",
    expected_value_percenage: "21.43",
  },
  {
    _id: "6357265b1e58f8dd69329d45",
    userId: "630ff7a767ae1e8f6a17631b",
    name: "Férias",
    total_value: 1000,
    initial_date: "2022-01-01T00:00:00.000Z",
    final_date: "2022-11-01T00:00:00.000Z",
    total_installments: 10,
    monthly_profitability: 0.72,
    invest_in_month: 4000,
    status: true,
    __v: 0,
    actual_value: 10000,
    actual_months: 12,
    remaining_months: -2,
    actual_value_percentage: "1000.00",
    expected_value_percenage: "120.00",
  },
  {
    _id: "63572785cdb7e7aacaff8934",
    userId: "630ff7a767ae1e8f6a17631b",
    name: "Imóvel",
    total_value: 400000,
    initial_date: "2022-01-01T00:00:00.000Z",
    final_date: "2026-01-01T00:00:00.000Z",
    total_installments: 72,
    monthly_profitability: 0.72,
    invest_in_month: 4000,
    status: true,
    __v: 0,
    actual_value: 420.5,
    actual_months: 12,
    remaining_months: 35,
    actual_value_percentage: "0.11",
    expected_value_percenage: "16.67",
  },
];

describe("Goal List", () => {
  it("should generate snapshot", () => {
    const { container } = render(<GoalList goals={goals} />);
    expect(container).toMatchSnapshot();
  });

  it("should generate table", () => {
    render(<GoalList goals={goals} />);
    const rowTable = screen.getAllByTestId("card-item");
    expect(rowTable.length).toBe(3);
  });
});
