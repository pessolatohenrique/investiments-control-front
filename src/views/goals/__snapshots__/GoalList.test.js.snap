// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goal List should generate snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        data-testid="card-item"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
          >
            Imóvel novo
          </div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            variant="body2"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Porcentagem atual: 
                  </strong>
                  5.69
                   %
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Porcentagem esperada: 
                  </strong>
                  21.43
                   %
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Valor total: 
                  </strong>
                  R$ 
                  400000
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Valor atual: 
                  </strong>
                  R$ 
                  22753.4
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Meses restantes: 
                  </strong>
                  47
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Meses atuais: 
                  </strong>
                  15
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Total de meses: 
                  </strong>
                  70
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Data de início: 
                  </strong>
                   
                  30/09/2021
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Data de fim: 
                  </strong>
                   
                  31/12/2026
                </p>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        data-testid="card-item"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
          >
            Férias
          </div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            variant="body2"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Porcentagem atual: 
                  </strong>
                  1000.00
                   %
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Porcentagem esperada: 
                  </strong>
                  120.00
                   %
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Valor total: 
                  </strong>
                  R$ 
                  1000
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Valor atual: 
                  </strong>
                  R$ 
                  10000
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Meses restantes: 
                  </strong>
                  -2
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Meses atuais: 
                  </strong>
                  12
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Total de meses: 
                  </strong>
                  10
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Data de início: 
                  </strong>
                   
                  31/12/2021
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Data de fim: 
                  </strong>
                   
                  31/10/2022
                </p>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        data-testid="card-item"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
          >
            Imóvel
          </div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            variant="body2"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Porcentagem atual: 
                  </strong>
                  0.11
                   %
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Porcentagem esperada: 
                  </strong>
                  16.67
                   %
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Valor total: 
                  </strong>
                  R$ 
                  400000
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Valor atual: 
                  </strong>
                  R$ 
                  420.5
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Meses restantes: 
                  </strong>
                  35
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Meses atuais: 
                  </strong>
                  12
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Total de meses: 
                  </strong>
                  72
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Data de início: 
                  </strong>
                   
                  31/12/2021
                </p>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                >
                  <strong>
                    Data de fim: 
                  </strong>
                   
                  31/12/2025
                </p>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
